app/javascript/controllers/app/search_controller.js

```diff
@@ -74,6 +78,12 @@ export default class extends Controller {
       return
     }
 
+    if (this.lastQuery === query) {
+      return
+    }
+
+    this.lastQuery = query
```

Comment:
What's this for?

Reply:
Per the commit message from 2a4ec283c1a1f70971b271d5d485b3b866d2bec3:

> **Don't fetch new search results if the query wasn't modified**
> 
> This is useful since this method will still be invoked when, for
example, the user presses the arrow keys or selects the text with
Control+A. In these cases fetching more results is unnecessary so let's
exit early.

Reply:
That makes total sense to me. Thanks!